lib/src/main/scala/ai/snips/bsonmacros/Macros.scala

```diff
@@ -154,7 +169,27 @@ class SetCodec[T](inner: Codec[T]) extends Codec[Set[T]] {
 	}
 }
 
-class MapCodec[A, B](inner: Codec[Any])(implicit ct: ClassTag[A]) extends Codec[Map[A, B]] {
+class ListCodec[T](inner: Codec[T]) extends Codec[List[T]] {
```

Comment:
I think we can start doing some refactoring on this one, the code for encoding/decoding is nearly the same for `ListCodec` / `SeqCodec` / `SetCodec`. A trait `IterableCodec` is suitable for handling encoding/decoding

Reply:
What about some simple refactoring like that: 

```
trait IterableCodec[T] {
  def encode(inner: Codec[T],
                writer: BsonWriter,
                it: Iterable[T],
                encoderContext: EncoderContext) {
    writer.writeStartArray()
    it.foreach(inner.encode(writer, _, encoderContext))
    writer.writeEndArray()
  }

  def decode(inner: Codec[T],
                reader: BsonReader,
                decoderContext: DecoderContext): Iterable[T] = {
    reader.readStartArray()
    val buffer = scala.collection.mutable.Buffer[T]()
    while (reader.readBsonType != BsonType.END_OF_DOCUMENT) {
      buffer.append(inner.decode(reader, decoderContext))
    }
    reader.readEndArray()
    buffer
  }
}

class SeqCodec[T](inner: Codec[T]) extends Codec[Seq[T]] with IterableCodec[T] {
  def getEncoderClass: Class[Seq[T]] = classOf[Seq[T]]

  def encode(writer: BsonWriter, it: Seq[T], encoderContext: EncoderContext) {
    encode(inner, writer, it, encoderContext)
  }

  def decode(reader: BsonReader, decoderContext: DecoderContext): Seq[T] = {
    decode(inner, reader, decoderContext).toSeq
  }
}
```

`IterableCodec` is not a real codec (perhaps we have to change is name) but just a trait which factorize `encoding`/`decoding` process for an iterable